src/Comps/cards/Cards.tsx

```diff
@@ -25,31 +31,32 @@ const Cards: React.FC<CardsProps> = ({ query, endpoint }) => {
 
   return (
     <CardsContainer>
-      {data.articles.map(
-        (
-          article: {
-            description: string;
-            author: string;
-            publishedAt: string;
-            urlToImage: string;
-            url: string | undefined;
-            title: string | undefined;
-          },
-          i: number
-        ) => {
-          return (
-            <Card
-              key={i}
-              description={article.description || ""}
-              author={article.author || ""}
-              publishedAt={changeDateForamt(article.publishedAt)}
-              urlToImage={article.urlToImage || ""}
-              urlToNews={article.url}
-              title={article.title}
-            ></Card>
-          );
-        }
-      )}
+      {data &&
+        data?.articles?.map(
+          (
+            article: {
+              description: string;
+              author: string;
+              publishedAt: string;
+              urlToImage: string;
+              url: string | undefined;
+              title: string | undefined;
+            },
+            i: number
+          ) => {
+            return (
+              <Card
+                key={i}
+                description={article.description || ""}
+                author={article.author || ""}
+                publishedAt={changeDateForamt(article.publishedAt)}
+                urlToImage={article.urlToImage || ""}
+                urlToNews={article.url}
+                title={article.title}
+              ></Card>
```

Comment:
if you don't need to render children, self close the card:
i.e. 
```suggestion
              />
```